### ISSUE TITLE
Support multiple hosting

### ISSUE THREAD

**Issue by hoang-innomizetech** [aws-amplify/amplify-cli#2022]
I am trying to add another hosting to my current repository, but I am getting below error
```
Error: Hosting is already fully enabled.
```

It would nice if Amplify supports multiple hosting so that we can add/remove/update multiple sites

**Comment by kaustavghosh06** [comment]
@hoang-innomizetech  Amplify has support for multiple environments if youre looking for that. The CLI is project specific for an environment - so I'm not sure if I get the use-case. Could you pleasep rovide more details?

**Comment by swaminator** [comment]
@hoang-innomizetech you can also use Amplify Console to host your frontends: https://console.amplify.aws

**Comment by hoang-innomizetech** [comment]
@kaustavghosh06 I wanted to configure multiple apps per a Monorepo. For example
* app1.domain.com point to app1.domain.com bucket
* app2.domain.com point to app2.domain.com bucket

All the above app should have its environments such as prod, staging or dev.

**Comment by hoang-innomizetech** [comment]
@swaminator our requirement is to configure multiple apps per Monorepo so that we can implement the auto-deployment solution

**Comment by kaustavghosh06** [comment]
@hoang-innomizetech At this moment the CLI doesn't support mono-repos and multiple apps with one Amplify CLI project.

**Comment by hoang-innomizetech** [comment]
@kaustavghosh06 yeah, I know that Amplify doesn't support it and asking for help.

Does anyone has any idea to use Amplify for hosting multiple apps per mono-repos?

**Comment by swaminator** [comment]
@hoang-innomizetech you can do that with the Amplify Console. You can connect the same repo to different apps, so you could have a React, Angular, Vue app all created from the same repo.

**Comment by hoang-innomizetech** [comment]
@swaminator  We don't want to setup apps manually on AWS console. However, I found another approach to setup hosting by using Serverless Website Component

**Comment by swaminator** [comment]
@hoang-innomizetech the service is named Amplify Console so can be confusing. But you can use Amplify's hosting offering via the AWS CLI, CloudFormation, or any of the AWS SDKs. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Amplify.html
https://docs.aws.amazon.com/cli/latest/reference/amplify/index.html

**Comment by eltonio450** [comment]
hi !

Is multiple hosting in the roadmap ? We also feel that the Amplify Console does not meet our needs in this case.

Thank you !

**Comment by julescsv** [comment]
This featured is greatly needed please!!!!!

**Comment by Bulletninja** [comment]
> @swaminator We don't want to setup apps manually on AWS console. However, I found another approach to setup hosting by using Serverless Website Component

source?

**Comment by dtelaroli** [comment]
+1

**Comment by mm87642** [comment]
+1

**Comment by msretin** [comment]
+1

**Comment by MontoyaAndres** [comment]
Any news?

**Comment by silveriordza** [comment]
Hello, I faced same question today,  I'm looking for 1 backend API supporting several FrontEnd websites using multiple amplify hosting, but I'm getting the "Amplify console hosting has already been enabled" alert when I try to add a 2nd hosting category, all on the same mono-repo,  this is because I have different flavors of the same main Front End, that have similar look and feel but are for different businesses, all sharing the same backend.

**Comment by royaldragonit** [comment]
`amplify pull --appId <your amplify Appid>--envName staging` will solve your error
